lib/screens/qr_screen.dart

```diff
@@ -0,0 +1,187 @@
+import 'dart:io';
+
+import 'package:cupertino_will_pop_scope/cupertino_will_pop_scope.dart';
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:qr_flutter/qr_flutter.dart';
+import 'package:qrdemo/providers/qr_provider.dart';
+import 'package:qrdemo/providers/timer_provider.dart';
+import 'package:qrdemo/screens/home_screen.dart';
+
+class QRScreen extends StatefulWidget {
+  const QRScreen({Key? key}) : super(key: key);
+
+  @override
+  _QRScreenState createState() => _QRScreenState();
+}
+
+class _QRScreenState extends State<QRScreen> {
+  @override
+  Widget build(BuildContext context) {
+    return Selector<QRProvider, String>(
+        selector: (_, provider) => provider.code!,
+        builder: (context, code, _) {
+          return QRImageView(data: code);
+        });
+  }
+}
+
+class QRImageView extends StatefulWidget {
+  const QRImageView({Key? key, this.data}) : super(key: key);
+
+  final String? data;
+
+  @override
+  _QRImageViewState createState() => _QRImageViewState();
+}
+
+class _QRImageViewState extends State<QRImageView> {
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return FutureBuilder<void>(
+      future: Provider.of<TimerProvider>(context, listen: false).startClock(),
```

Comment:
I like to make sure everything is prepared before building any UI so that all the contents will be available. The UI will be glitchy sometimes if I don't start the timer in time. I might get a null error because timer is null. This build method is only called when the user navigates to QRScreen. The reason why these build methods aren't refreshing is because I used a selector that listens for changes and the providers notifies the listener widgets to update the UI.